### DISCUSSION TITLE
Purpose of Deposit and Withdraw in FA/Coin

### DISCUSSION THREAD

**Discussion by kshitijc1506** [aptos-labs/aptos-developer-discussions#210] [Questions] [answered]
### Discord user ID

kc1506

### Describe your question in detail.

I have 2 questions regard withdraw and deposit:

1. Why use `withdraw` and `deposit` for Fungible Assets/Coins instead of a single transfer function like in ERC-20?
2. Since `deposit` is a public function callable by any module or script, how do we ensure that only authorized users can call `deposit`? Specifically, how do we prevent unauthorized users from increasing their fungible store balance by calling `deposit`?

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

**Comment by kshitijc1506** [comment]
From my research, here's what I've gathered:

1. `transfer` abstracts away individual steps, perhaps making it slightly easier to use. However, `deposit` and `withdraw` is more granular and flexible.  Moreover, with Dispatchable FA you can write custom logic in both steps (again, more granularity). However, still would like to know if there is an example I can use to drive this concept home (i.e. is there any specific moment one would like to use either `deposit` or `withdraw` without the typical transfer flow of calling one after the other?) @gregnazario @0xaptosj 

2. `deposit` functions take in a `FungibleAsset` parameter. This [struct](https://github.com/aptos-labs/aptos-core/blob/5770f4a16821aa120396bb3d862497fdeb39ec13/aptos-move/framework/aptos-framework/sources/fungible_asset.move#L153) is a part of the Metadata object. You cannot obtain this ephemeral struct without calling `withdraw` or `mint` first. Therefore, the given concern is invalid.

**Reply by gregnazario** [reply, marked as answer]
Yes, transfer is built on a withdraw from the sender and a deposit on the reciever.

Deposit and withdraw provide building blocks for pulling coins out and storing them in other resources.

For FungibleAsset, this paradigm is kept but it still needs to go to a store.

There's additionally some gas optimization you can do here, where you withdraw once and split the coin to send to multiple receivers. Similarly you can withdraw and burn tokens in that way
